packages/mui-material/src/Dialog/Dialog.d.ts

```diff
@@ -52,6 +52,10 @@ export interface DialogProps extends StandardProps<ModalProps, 'children'> {
    * @deprecated Use the `onClose` prop with the `reason` argument to handle the `backdropClick` events.
    */
   onBackdropClick?: ModalProps['onBackdropClick'];
+  /**
+   * Callback fired when the dialog is clicked.
+   */
+  onClick?: ModalProps['onClick'];
```

Comment:
It looks like we shouldn't document this per the API convention

```suggestion
  /**
   * @ignore
   */
  onClick?: ModalProps['onClick'];
```

Reply:
Thank you! I had a feeling I was missing something.